### DISCUSSION TITLE
Seeed Xiao Rp2040 and GC9A01 stop working after updates

### DISCUSSION THREAD

**Discussion by djairjr** [Bodmer/TFT_eSPI#2002] [Compatible support libraries]
Hi Bodmer. 
A few months ago, I´ve managed to run Animated Eyes example using the Seeed Xiao Rp2040 and two GC9A01 round displays.
https://www.pcbway.com/project/shareproject/Finally_Animated_Eyes_using_Seed_Xiao_RP2040_d7af7894.html

Now the same code does not work. I´ve made all upgrades in Erle core and the default Arduino for Rp2040 core. And use same UserSetup, but there is no image at all.

The Read_User_Setup example does not return the correct SPI Pinouts (used by Seeed Xiao Rp2040). Instead, return this:

[code]
TFT_eSPI ver = 2.4.72
Processor    = RP2040
Transactions = Yes
Interface    = SPI
Display driver = 9A01
Display width  = 240
Display height = 240

MOSI    = GPIO 3
SCK     = GPIO 2
TFT_CS   = GPIO 31
TFT_DC   = GPIO 6
TFT_RST  = GPIO 7

TFT_BL           = GPIO -1
TFT_BACKLIGHT_ON = HIGH

Font GLCD   loaded
Font 2      loaded
Font 4      loaded
Font 6      loaded
Font 7      loaded
Font 8      loaded
Smooth font enabled

Display SPI frequency = 27.00
[/code]

In my custom user setup, the pin definitions are as follows:

[code]
#define RP2040_PIO_SPI // Erle Core
#define GC9A01_DRIVER // Round TFT SPI Display 

#define TFT_WIDTH  240 // ST7789 240 x 240 and 240 x 320
#define TFT_HEIGHT 240 // GC9A01 240 x 240

#define TFT_SCLK SCK // PIN D8 VERDE
#define TFT_MOSI MOSI // TX PIN D10 AZUL 
#define TFT_CS   SS  // Chip select control PIN D7
// Second one is D6 Pin
#define TFT_DC   D4  // Data Command control PIN D4
#define TFT_RST  D5  // Reset pin 

#define TFT_BL   -1            // LED back-light control pin
#define TFT_BACKLIGHT_ON HIGH  // Level to turn ON back-light (HIGH or LOW)

#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts

#define SMOOTH_FONT
// #define PIXEL_DOUBLE 

//#define SPI_FREQUENCY  20000000
#define SPI_FREQUENCY  27000000
//#define SPI_FREQUENCY  40000000
//#define SPI_FREQUENCY  62500000
//#define SPI_FREQUENCY  80000000

// Optional reduced SPI frequency for reading TFT
#define SPI_READ_FREQUENCY  5000000
[/code]

**Comment by Bodmer** [comment]
Check the User_Setup_Select.h file is calling up the correct setup file. When this is correct then the Read_User_Setup sketch will report the defined pins set.

**Reply by djairjr** [reply]
My User_Setup_Select.h file is correct and pointing to the correct setup file.

**Comment by Bodmer** [comment]
Try commenting out this line thus:

// #define RP2040_PIO_SPI

The built in SPI port will then be used.

**Reply by djairjr** [reply]
Commenting this line does not make any changes.

**Reply by djairjr** [reply]
Bodmer, my first attempts to recover the operation of the system were precisely the ones you have recommended. I am repeating only for confirmation, intending to help solve the problem. 
In addition to the ones you recommend, I've already disabled DMA in the AnimatedEyes script itself. 
Commenting line 
`#define USE_DMA `
I also tried several speed settings of the SPI Bus. 
My next check will be regarding the boot sequence of the display itself.

**Comment by Bodmer** [comment]
Assuming you are trying to use the GPIO pins in #2000 then I tested my display with the following setup file and it is working fine:

```
#define RP2040_PIO_SPI

#define GC9A01_DRIVER

#define TFT_MOSI  10
#define TFT_MISO  9 // Not connected to display
#define TFT_SCLK  8
#define TFT_CS   7  // Chip select control pin
#define TFT_DC   4  // Data Command control pin
#define TFT_RST  5  // Reset pin (could connect to Arduino RESET pin)

#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:-.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts

#define SMOOTH_FONT

#define SPI_FREQUENCY  33000000
#define SPI_READ_FREQUENCY  20000000
#define SPI_TOUCH_FREQUENCY  2500000
```

I am using Phil's latest board package v 2.5.2
Not sure what to suggest.
